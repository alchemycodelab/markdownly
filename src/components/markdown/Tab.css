.Tab{
  border: 1px solid black;
  background-color: crimson;
  
  button{
    border: none;
    outline: none;
  }
  button:hover{
    background-color: pink;
  }
  .fileTab{
    background-color: crimson;
    min-width: 2em;
  }
  .deleteButton{
    background-color: crimson;
  }
}