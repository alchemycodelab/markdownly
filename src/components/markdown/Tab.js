import React from 'react';
import PropTypes from 'prop-types';
import styles from './Tab.css';

export const Tab = ({ title, handleClick, handleDelete, focus }) => {

  return (
    <span className={styles.Tab}>
      <button className={styles.fileTab} name={title} onClick={handleClick}>
        {title}
      </button>
      <button className={styles.deleteButton} name={title} onClick={()=>handleDelete(focus, title)}>âŒ«</button>
    </span>
  );
};

Tab.propTypes = {
  title: PropTypes.string.isRequired,
  handleClick: PropTypes.func.isRequired,
  handleDelete: PropTypes.func.isRequired,
  focus: PropTypes.number.isRequired
};
