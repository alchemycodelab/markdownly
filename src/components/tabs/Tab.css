.Tab {
  display: flex;
  h2 {
    font-size: .8em;
    margin: 0;
    padding: 5px;
  }
  button {
    font-family: Arial, Helvetica, sans-serif;
    width: 2em;
    height: 2em;
    font-size: .2em;
    padding: 0 .5em;
    margin: .5em;
    text-align: center;
    border: none;
  }
}