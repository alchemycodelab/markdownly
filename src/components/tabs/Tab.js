import React from 'react';
import PropTypes from 'prop-types';
import styles from './Tab.css';

const Tab = ({ title, changeTabs, deleteTab }) => {
  return (
    <div className={styles.Tab}>
      <h2 onClick={() => changeTabs(title)}>{title}</h2>
      <button onClick={() => deleteTab(title)}>ðŸ—‘</button>
    </div>
  );
};

Tab.propTypes = {
  title: PropTypes.string,
  changeTabs: PropTypes.func,
  deleteTab: PropTypes.func
};

export default Tab;
