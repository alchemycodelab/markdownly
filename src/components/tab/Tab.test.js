import React from 'react';
import { shallow } from 'enzyme';
import Tab from './Tab';

describe('Tab component', () => {
  it('renders Tab', () => {
    const wrapper = shallow(<Tab />);
    expect(wrapper).toMatchSnapshot();
  });
});
