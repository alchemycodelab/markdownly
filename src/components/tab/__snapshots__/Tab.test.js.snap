// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tab component renders Tab 1`] = `
<li>
  <input
    name="tabs"
    type="radio"
  />
  <label>
     TAB 
  </label>
</li>
`;
